packages/features/pbac/services/role.service.ts

```diff
@@ -0,0 +1,145 @@
+import type { PrismaClient } from "@prisma/client";
+
+import type { PermissionString, Resource, Action } from "../types/permission-registry";
+import { PermissionService } from "./permission.service";
+
+export class RoleService {
+  private prisma: PrismaClient;
+  private permissionService: PermissionService;
+
+  constructor(prisma: PrismaClient) {
+    this.prisma = prisma;
+    this.permissionService = new PermissionService();
+  }
+
+  async createRole(data: {
+    name: string;
+    description?: string;
+    teamId?: number;
+    permissions: PermissionString[];
+  }) {
+    // Validate permissions
+    if (!this.permissionService.validatePermissions(data.permissions)) {
+      throw new Error("Invalid permissions provided");
+    }
+
+    return this.prisma.$transaction(async (tx) => {
```

Comment:
Wrapping this in a transaciton because this failed in my testing and broke my dev instance since roles permissions are unique to roles